CMakeLists.txt

```
@@ -2236,13 +2238,17 @@ add_hpx_library_headers_noglob(hpx_external)
 include(HPX_SetupMPI) # must come before APEX
 
 # Setup packages and subprojects
+include(HPX_SetupHIP)
 include(HPX_SetupCUDA) # CUDA needs to come before Apex, to have
                        # cuda_add_library available
 include(HPX_SetupApex)
 include(HPX_SetupPapi)
 include(HPX_SetupHpxmp)
 include(HPX_SetupGooglePerfTools)
 include(HPX_SetupValgrind)
+if(HPX_WITH_CUDA OR HPX_WITH_HIP)
+  hpx_add_config_define(HPX_HAVE_GPU_SUPPORT)
```

```suggestion
  hpx_add_config_define(HPX_HAVE_GPU)
```
just for consistency? We don't tend to add `SUPPORT` to any of the options or defines.

---

`HPX_WITH_GPU` cmake option and `HPX_HAVE_GPU_SUPPORT` define have different purpose (I should probably change names if it is not that clear):
- `HPX_WITH_GPU` cmake option enables both `async_cuda` and `compute_cuda` compilation
- `HPX_HAVE_GPU_SUPPORT` is defined when either `HPX_HAVE_CUDA` or `HPX_HAVE_HIP`, which can be defined if `HPX_WITH_GPU` is not on (in case just `async_cuda` module is enabled)

I'm open to a recfactoring if you come with a better idea ^^'

---

So in the end:
-  I'll remove the `HPX_WITH_GPU` cmake option
- Both `HPX_WITH_ASYNC_CUDA` and `HPX_WITH_COMPUTE_CUDA` (to be renamed in a future PR) will be set to `ON` by default
- If hipcc compiler is not used and those options are on, the cuda implementation will be chosen by default